File: cdk/typescript/route53/lib/route53-stack.ts
import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import {
  Vpc,
  SubnetType,
  IpAddresses,
} from 'aws-cdk-lib/aws-ec2';
import * as route53 from 'aws-cdk-lib/aws-route53';

export class Route53Stack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    // Create private VPC with cidr block 10.0.0.0/16
    const vpc = new Vpc(this, 'RdsAuroraMysqlVpc', {
      ipAddresses: IpAddresses.cidr('10.0.0.0/16'),
      restrictDefaultSecurityGroup: false,
      natGateways: 0,
      subnetConfiguration: [
        {
          cidrMask: 24,
          name: 'private',
          subnetType: SubnetType.PRIVATE_ISOLATED,
        }
      ]
    });

    // Create a public hosted zone "mydomain.com"
    new route53.PublicHostedZone(this, 'route53PublicHostedZone', {
      zoneName: 'mydomain.com',
    });

    // Create a private hosted zone "local.mydomain.com" for the vpc
    new route53.PrivateHostedZone(this, 'route53PrivateHostedZone', {
      vpc: vpc,
      zoneName: 'local.mydomain.com',
    });

  }
}

File: cdk/python/route53/route53/route53_stack.py
from aws_cdk import (
    Stack,
    aws_route53 as route53,
    aws_ec2 as ec2,
)
from constructs import Construct

class Route53Stack(Stack):

    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
        super().__init__(scope, construct_id, **kwargs)

        # Create basic VPC with cidr block 10.0.0.0/16
        vpc = ec2.Vpc(
            self,
            "VPC",
            ip_addresses=ec2.IpAddresses.cidr("10.0.0.0/16"),
            restrict_default_security_group=False,
            max_azs=2,
            nat_gateways=0,
            subnet_configuration=[
                ec2.SubnetConfiguration(
                    subnet_type=ec2.SubnetType.PRIVATE_ISOLATED,
                    cidr_mask=24,
                    name="PrivateSubnet"
                ),
            ],
        )

        # Create a public hosted zone
        hosted_zone = route53.PublicHostedZone(
            self,
            "PublicHostedZone",
            zone_name="mytest.domain.com",
        )

        # Create a private hosted zone
        private_hosted_zone = route53.PrivateHostedZone(
            self,
            "PrivateHostedZone",
            zone_name="private.mytest.domain.com",
            vpc=vpc,
        )

